vignettes/articles/glitter_for_Wikidata.Rmd

```diff
@@ -18,178 +18,225 @@ knitr::opts_chunk$set(
 
-You could access this information through:
+# Use glitter functions to start exploring data
 
-```{r}
+## Simple single-line query
+
+The `glitter` functions might now be used to start exploring data.
+
+We're looking for items (the "unknown" in our query below, hence the use of a "?") which are part of the Lyon metro network:
+
+```{r single_line}
 stations=get_triple("?items wdt:P16 wd:Q1552")
-stations %>% head()
+head(stations)
 ```
 
-Notice that we do not have values yet for the stations (that's what we're looking for) hence the use of "?" at the beginning of the subject string.
-
 To also get the labels for stations, we can use the argument `label`:
 
-```{r}
-parts_metro_Lyon=get_triple("?items wdt:P16 wd:Q1552", label="?items")
-parts_metro_Lyon %>% head()
+```{r single_line_with_label}
+parts_metro_Lyon=get_triple("?items wdt:P16 wd:Q1552",
+                            label="items")
+head(parts_metro_Lyon)
 ```
 
-For now, we get 50 items, not only stations but also other types of items such as metro lines. Let's have a look at the item "Place Guichard - Bourse du Travail" (Q59855) which we know correspond to a station.
-
-Property "wdt:P31" should enable us to collect only stations ("wd:Q928830") instead of all parts of the Lyon Metro network.
+## Simple multi-line query
 
-We can enrich and refine our query incrementally with `spq_filter()` before sending it (whereas `get_triple()` was a shortcut to build and send simple requests) :
+The `get_triple()` function we used above is a good option when you want to send **a very simple query**, yet more often than not one will wish to **gradually complexify** their initial query, so that it's actually preferable to use the **multi-line equivalent** to get_triple, which allows for such increments:
 
-```{r}
-stations_metro_Lyon=spq_init() %>% 
-  spq_add("?stations wdt:P361 wd:Q1552", .label = "?stations") %>% 
-  spq_add("?stations wdt:P31 wd:Q928830") %>% 
+```{r multi_line}
+parts_metro_Lyon=spq_init() %>% 
+  spq_add("?items wdt:P16 wd:Q1552") %>% 
+  spq_label("items") %>% 
   spq_perform()
-head(stations_metro_Lyon)
+dim(parts_metro_Lyon)
+head(parts_metro_Lyon)
 ```
 
-Or we could use glitter DSL (domain-specific language) a bit more, making use of `spq_set()` and the `spq_filter()` fonction.
+## Labelling
 
-```{r}
-stations_metro_Lyon = spq_init() %>% 
-  spq_set(stations_code = "wd:Q1552") %>%
-  spq_filter(stations == wdt::P361(stations_code), .label="?stations") %>% 
-  spq_set(lyon_code = "wd:Q928830") %>% 
-  spq_filter(stations == wdt::P31(lyon_code)) %>% 
-  spq_select(- stations_code, - lyon_code) %>%
+The query above, with `spq_label(?items)`, will return a table comprising both `items` (with the Wikidata identifiers) and `itemsLabel` (with the human-readable label corresponding to these items). If the Wikidata unique identifier is not particularly useful, one can use the argument `.overwrite=FALSE` so that only labels will be returned, under the shorter name `items`:
```

Comment:
```suggestion
The query above, with `spq_label(items)`, will return a table comprising both `items` (with the Wikidata identifiers) and `itemsLabel` (with the human-readable label corresponding to these items). If the Wikidata unique identifier is not particularly useful, one can use the argument `.overwrite=TRUE` so that only labels will be returned, under the shorter name `items`:
```

Reply:
oh, yes. And also, I think it's items_label and not itemsLabel now